syntax = "proto3";

option java_package = "ex.grpc";
option objc_class_prefix = "HSW";

package example;

service ProductService {
  rpc ListProduct (ListRequest) returns (ListReply) {}
  rpc GetCatalogue (CatalogueRequest) returns (CatalogueReply) {}
}

message ListRequest {
  string product_name = 1;
  int32 seller_id = 2;
  double price = 3;
}

message ListReply {
  string message = 1;
}

message CatalogueRequest {
  int32 next_product_id = 1;
  int32 limit = 2;
}

message Product {
  int32 product_id = 1;
  string product_name = 2;
  double product_price = 3;
  string seller_name = 4;
}

message CatalogueReply {  
  repeated Product products = 1;
  int32 next_product_id = 2;
}
